### ISSUE TITLE
Refactoring and cleaning up Shoes eBook

### ISSUE THREAD

**Issue by IanTrudel** [Shoes3/ebook#17]
Please, list refactoring and cleaning up tasks relevant to the release of Shoes eBook.

- Directory structure and filenames should be reviewed. 
- Get rid of files that were used to start the project but are not used anymore.
- Refactor and simplify code wherever it is possible (eBook feels a tad bloated already).
- Name convention would need some work.

**Comment by IanTrudel** [comment]
Thanks for cleaning up.

**Comment by ccoupe** [comment]
ebook-builder is a big script and it won't get any smaller in the future as features are added.  With Shoes, it's hard  to separate gui  from processing done at the save button and questionable if doing so improves understanding.

**Comment by IanTrudel** [comment]
Indeed, it's a huge challenge. There are ways to separate logic and UI but may not improve readability.

Numinoes is implemented with a MVC pattern. The eBook builder might not gain from this approach.

Some refactoring is possible.

It would also be possible to segment the code into multiple files: eBook-builder.rb, eBook-builder-init-button.rb, eBook-builder-preprocessing-button.rb, etc.

Just few ideas here. It's probably best to wait a bit before doing anything.

**Comment by IanTrudel** [comment]
The expert Shoes samples all define their own class holding the logic, no significant logic within `Shoes.app`.

**Comment by ccoupe** [comment]
A good first step might be to create a class that abstracts the @cfg manipulations and perhaps allow a gdbm or sqlite data store.  That's far, far, far away. We haven't even nailed down what the yaml/gdbm/sqlite3 data store has much less the proper methods to call to get what is needed.
